client/my-sites/post-type-list/post-actions-ellipsis-menu/qrcode.jsx

```diff
@@ -0,0 +1,65 @@
+					fill-color="#000000"
+				/>
+			</svg>
+		}
+	/>
+);
+
+function PostActionsEllipsisMenuQRCode( { translate, status, url } ) {
+	const [ showQRCode, setShowQRCode ] = useState( false );
+
+	const generateQRCode = () => {
+		setShowQRCode( true );
+	};
+
+	if ( ! url || ! status ) {
+		return null;
+	}
+
+	// The QR option will be available only for `publish` status
+	if ( status !== 'publish' ) {
+		return null;
+	}
+
+	return (
+		<>
+			<PopoverMenuItem onClick={ generateQRCode } icon={ qrIcon }>
+				{ translate( 'QR Code' ) }
+			</PopoverMenuItem>
+			<PostActionsQRCode siteUrl={ url } showQRCode={ showQRCode } />
+		</>
+	);
+}
+
+PostActionsEllipsisMenuQRCode.propTypes = {
+	globalId: PropTypes.string.isRequired,
+};
+
+const mapStateToProps = ( state, { globalId } ) => {
```

Comment:
Using `connect()` is not something we generally do with functional components. Instead, it's preferred that we use the `useSelector()` hook directly in the functional component. That should also help simplify the code a bit.

Reply:
Thanks for the recommendation! You are right. Just refactored the component to use `useSelector()`.